### PR TITLE
Improve test by waiting for safe state before destroy [HZ-1849]

### PR DESCRIPTION
**Pull request by ahmetmircik**
closes https://github.com/hazelcast/hazelcast/issues/19834

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/map/LocalIndexStatsTest.java [hunk 1]
```diff
@@ -635,6 +635,7 @@ public void testIndexStatsAfterMapDestroy() {
         assertTrue(valueStats().getTotalUpdateLatency() > 0);
         assertTrue(valueStats().getAverageHitLatency() > 0);
 
+        ensureSafeState();
```
Comment (mdumandag) [inline hazelcast/src/test/java/com/hazelcast/map/LocalIndexStatsTest.java:638]: Can you explain how this helps with this test failure?

Comment (ahmetmircik) [reply hazelcast/src/test/java/com/hazelcast/map/LocalIndexStatsTest.java:638]: destroy is not a partition operation, to be sure there is no race condition with any other partition operation, i have put that safe-state check before destroy. It is useful to eliminate one more possible case when looking into next failure.

Comment (mdumandag) [reply hazelcast/src/test/java/com/hazelcast/map/LocalIndexStatsTest.java:638]: OK, let's try this

**Review by mdumandag — APPROVED** [review]

**Comment by ahmetmircik** [comment]
simple test improvement merging with 1 ok.

**Pull request merged**